libraries/audio/impl/src/main/kotlin/io/element/android/libraries/audio/impl/DefaultAudioFocus.kt

```diff
@@ -30,18 +30,34 @@ class DefaultAudioFocus @Inject constructor(
     private var audioFocusChangeListener: AudioManager.OnAudioFocusChangeListener? = null
 
     @Suppress("DEPRECATION")
-    override fun requestAudioFocus(mode: AudioFocusRequester) {
+    override fun requestAudioFocus(
+        mode: AudioFocusRequester,
+        onFocusLost: () -> Unit,
+    ) {
+        val listener = AudioManager.OnAudioFocusChangeListener {
+            when (it) {
+                AudioManager.AUDIOFOCUS_GAIN -> {
+                    // Do nothing
+                }
+                AudioManager.AUDIOFOCUS_LOSS,
+                AudioManager.AUDIOFOCUS_LOSS_TRANSIENT,
+                AudioManager.AUDIOFOCUS_LOSS_TRANSIENT_CAN_DUCK -> {
```

Comment:
Maybe ignore it for the `_CAN_DUCK` variant if the requester is the media viewer? https://developer.android.com/media/optimize/audio-focus#automatic-ducking

Apparently the OS will automatically duck the audio in that case, but we shouldn't do it for voice-related audio (messages, call).

Reply:
Reading the doc again, I think I can improve the code behavior yes. I will add more commits.

Reply:
OK, so I was about to check the result of the audio focus requester to prevent the start of the play back, but it seems that (well, on my phone at least) when there is an active call, the audio is ducked automatically by the system. Other apps does not always prevent playback during a phone call, for instance:
- Netflix: cannot play
- Youtube and Spotify: can start playback, but ducked.

So without doing anything, Element will have the same behavior than Youtube and Spotify, which is probably acceptable.